scripts/patch_hosts.py

```diff
@@ -0,0 +1,212 @@
+# stdlib
+import argparse
+from functools import cached_property
+import os
+from pathlib import Path
+import platform
+import re
+import sys
+
+
+class Platform:
+    system = platform.system()
+    uname = platform.uname()
+
+    @staticmethod
+    def get() -> str:
+        return Platform.system
+
+    @staticmethod
+    def windows() -> bool:
+        return Platform.system == "Windows"
+
+    @staticmethod
+    def macos() -> bool:
+        return Platform.system == "Darwin"
+
+    @staticmethod
+    def linux() -> bool:
+        return Platform.system == "Linux"
+
+    @staticmethod
+    def wsl2() -> bool:
+        return "wsl2" in Platform.uname.release.lower()
+
+
+class Hosts:
+    def __init__(self, path: str = None) -> None:
+        self.__path = path
+        self.content = self.read()
+
+    @cached_property
+    def path(self) -> Path:
+        if self.__path:
+            p = self.__path
+        elif Platform.linux():
+            p = "/etc/hosts"
+        elif Platform.macos():
+            new_hosts = "/etc/hosts"
+            old_hosts = "/private/etc/hosts"
+            p = new_hosts if os.path.exists(new_hosts) else old_hosts
+        elif Platform.windows():
+            p = "C:\Windows\System32\drivers\etc\hosts"
+        else:
+            raise Exception(f"Unsupported OS: {Platform.system()}")
```

Comment:
NIT: Platform.system is a string, we could remove the brackets

Reply:
eesh this would've caused an exception. pushed a fix.